x/subscription/keeper/subscription.go

```diff
@@ -252,7 +228,6 @@ func (k Keeper) upgradeSubscriptionPlan(ctx sdk.Context, duration uint64, sub *t
 	// The "old" subscription's duration is now expired
 	// If called from CreateSubscription, the duration will reset to the duration bought
 	sub.DurationLeft = 0
-	sub.Cluster = types.GetClusterKey(*sub)
```

Comment:
why?

Reply:
Because we already do it later in the function `resetSubscriptionDetailsAndAppendEntry`